web/js/playground.js

```diff
@@ -93,18 +101,33 @@ function renderTools() {
     toolSetContainer.innerHTML = '';
 
     const recentTools = getRecentlyUsedTools();
+    const hasRecentlyUsedTools = recentTools.length > 0;
 
-    // Sort tools by their recent usage
-    const sortedTools = [...ToolsCatalog].sort((a, b) => {
-        const indexA = recentTools.findIndex(tool => tool.label.toLowerCase() === a.label.toLowerCase());
-        const indexB = recentTools.findIndex(tool => tool.label.toLowerCase() === b.label.toLowerCase());
-        return indexA === -1 ? 1 : indexB === -1 ? -1 : indexA - indexB;
-    });
+    // Divide ToolsCatalog into recently used tools and non-recently-used tools
+    const recentlyUsedLabels = recentTools.map(tool => tool.label.toLowerCase());
+    const recentToolsSet = new Set(recentlyUsedLabels);
```

Comment:
Is this portion of the code creating new objects for storing both the list of recently used and unused tools before recombining them in order to sort the whole list?